Readme.md

```diff
@@ -0,0 +1,494 @@
+        database.runCommand(command)
+        println("Pinged your deployment. You successfully connected to MongoDB!")
+        database
+    } catch (me: MongoException) {
+        System.err.println(me)
+        null
+    }
+}
+```
+
+> An additional check for connectionString is added to demo, allowing you to use connection URI directly for ease, but it is strongly recommended
+> to avoid hardcoding of connection URI.
+
+With `setupConnection` function ready, let's test it, and query the database for collection count and name.
+
+```kotlin
+suspend fun listAllCollection(database: MongoDatabase) {
+
+    val count = database.listCollectionNames().count()
+    println("Collection count $count")
+
+    print("Collection in this database are -----------> ")
+    database.listCollectionNames().collect { print(" $it") }
+}
+```
+
+And on running our output would look like this.
+
+![list collection output](https://images.contentstack.io/v3/assets/blt39790b633ee0d5a7/blt5a670a8008abba48/648835185953929729a04668/Screenshot_2023-06-13_at_10.21.15.png)
+
+And you would have noticed by now that we are using `suspend` keyword here with `listAllCollection()`. Since `listCollectionNames()` interact with
+the database it is recommended to run it on a different thread and since MongoDB Kotlin driver supports the native Kotlin asynchronous language
+paradigm i.e. [Coroutines](https://kotlinlang.org/docs/coroutines-guide.html) ,and we can benefit from it by using `suspend` functions.
+
+Similarly, we can drop collections using `suspend` function.
+
+```kotlin
+suspend fun dropCollection(database: MongoDatabase) {
+    database.getCollection<Objects>(collectionName = "test").drop()
+}
+```
+
+With this complete, we are all set to start working on our CRUD application. So to start with, we need to create a `data` class that represents
+restaurant information which would be saved into the database.
+
+```kotlin
+data class Restaurant(
+    @BsonId
+    val id: ObjectId,
+    val address: Address,
+    val borough: String,
+    val cuisine: String,
+    val grades: List<Grade>,
+    val name: String,
+    @BsonProperty("restaurant_id")
+    val restaurantId: String
+)
+
+data class Address(
+    val building: String,
+    val street: String,
+    val zipcode: String,
+    val coord: List<Double>
+)
+
+data class Grade(
+    val date: Date,
+    val grade: String,
+    val score: Int
+)
+```
+
+In the above code snippet, we used two annotations
+
+1. `@BsonId` which are used to represent to the as unique identity `_id` of a document.
+2. `@BsonProperty` which is used to create an alias for keys in the document like this case it represents `restaurant_id`.
+
+> Also, to note here our `Restaurant` data class here is exact replicate of restaurant document, but we can skip few fields like `grades` or
+`address` or make them optional while maintaining the ability to perform CRUD operation over it as MongoDB support flexible schema.
```

Comment:
Does an ignore happen by default in Kotlin? In C# you have to explicitly state which fields you want to ignore to prevent an error.

Reply:
Sorry, I think so you understood it around way round.  We don't need to have explicate of document into the code to perform crud operation. 

`@Ignore ` is think so what you are talking about which allows you to ignore fields in syncing.